TeamCode/src/main/java/org/firstinspires/ftc/teamcode/api/linear/Encoders.kt

```diff
@@ -0,0 +1,135 @@
+package org.firstinspires.ftc.teamcode.api.linear
+
+import com.qualcomm.robotcore.hardware.DcMotor
+import com.qualcomm.robotcore.hardware.DcMotorSimple
+import org.firstinspires.ftc.teamcode.api.TriWheels
+import kotlin.math.abs
+
+/**
+ * Requires the [TriWheels] API.
+ */
+object Encoders : LinearAPI() {
+    private const val TICKS_PER_INCH: Double = 39.0
+    private const val TICKS_PER_DEGREE: Double = 6.5
+
+    private const val ENCODER_GAIN: Double = 0.0004
+    private const val ENCODER_ERROR: Int = 10
+
+    private fun inchesToTick(inches: Double): Int = (this.TICKS_PER_INCH * inches).toInt()
+    private fun degreesToTick(degrees: Double): Int = (this.TICKS_PER_DEGREE * degrees).toInt()
+
+    /**
+     * Drives the robot in a given [direction] a certain amount of [inches].
+     *
+     * This function will take full control of the robot, and has a few side-effects. All wheel
+     * motors will be reset with [TriWheels.stopAndResetMotors]. Furthermore, this function will not
+     * return until the robot has finished moving.
+     */
+    fun driveTo(direction: Direction, inches: Double) {
+        TriWheels.stopAndResetMotors()
+
+        val (left, right, _) = this.defineWheels(direction)
+        val ticks = this.inchesToTick(inches)
+
+        val leftTarget = left.currentPosition + ticks
+        val rightTarget = right.currentPosition + ticks
+
+        // Set left direction to reverse, undoing it at the end even if an exception is thrown
+        try {
+            right.direction = DcMotorSimple.Direction.REVERSE
+
+            while (
+                abs(left.currentPosition - leftTarget) > this.ENCODER_ERROR &&
+                abs(right.currentPosition - rightTarget) > this.ENCODER_ERROR &&
+                linearOpMode.opModeIsActive()
+            ) {
+                left.power = abs(left.currentPosition - leftTarget) * this.ENCODER_GAIN
+                right.power = abs(right.currentPosition - rightTarget) * this.ENCODER_GAIN
```

Comment:
This code only allows the robot to go forward along that direction. Consider testing alternative methods for this logic?

Reply:
I have a local fix that solves this problem, but again it needs testing before I push it.